### ISSUE TITLE
Celery logger stops working using torch.hub.load

### ISSUE THREAD

**Issue by arjitkatare** [ultralytics/yolov5#6060]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

PyTorch Hub

### Bug

For some reasons, while using torch.hub.load in celery worker, logger is getting shutdown. By specifying v6.0 in repo_or_dir, issue seems to be resolved

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@arjitkatare 👋 hi, thanks for letting us know about this possible problem with YOLOv5 🚀. We've created a few short guidelines below to help users provide what we need in order to get started investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub [master](https://github.com/ultralytics/yolov5/tree/master), and if necessary `git pull` or `git clone` a new copy to ensure your problem has not already been solved in [master](https://github.com/ultralytics/yolov5/tree/master).
- ✅ **Unmodified** – Your problem must be reproducible using official YOLOv5 code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/yolov5/issues/new/choose) with a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem. 

Thank you! 😃

**Comment by JonathanSamelson** [comment]
I ran into the same issue with my own loggers after updating PyTorch to 1.10 (from 1.7.1).

The problem comes from this line:
`self.net = torch.hub.load('ultralytics/yolov5', 'custom', path=self.model_path, verbose=False)`

In this case, the output of my script is as follows:
```
2022-03-11 17:42:03 pluton aptitude-toolbox[22744] INFO Model type YOLO selected.
YOLOv5  2022-1-12 torch 1.10.2+cu113 CUDA:0 (GeForce GTX 1080 Ti, 11264MiB)

Fusing layers...
Model Summary: 213 layers, 7039792 parameters, 0 gradients
Adding AutoShape...
Detector init duration = 4.6338949000000005s
Model type SORT selected.
Tracker init duration = 0.019383399999999718s
```

Then, I don't have any output from my loggers anymore.

Whereas when adding a tag such as v6.0:
`self.net = torch.hub.load('ultralytics/yolov5:v6.0', 'custom', path=self.model_path, verbose=False)`

My loggers are working and don't stop:
```
2022-03-11 17:46:15 pluton aptitude-toolbox[12680] INFO Model type YOLO selected.
2022-03-11 17:46:19 pluton aptitude-toolbox[12680] INFO Detector init duration = 4.442983099999999s
2022-03-11 17:46:19 pluton aptitude-toolbox[12680] INFO Model type SORT selected.
2022-03-11 17:46:19 pluton aptitude-toolbox[12680] INFO Tracker init duration = 0.019554799999999872s
.... [After the process ends] ...
2022-03-11 17:46:21 pluton aptitude-toolbox[12680] INFO Average FPS: 12.473039525066566
2022-03-11 17:46:21 pluton aptitude-toolbox[12680] INFO Average FPS w/o read time: 13.190384737142026
```

As you can see, I still have the details provided by my logger in the latter case.

Also, using the latest tag (v6.1) give me another error, which is not related I think:
`Exception: path is on mount 'C:', start on mount 'E:'. Cache may be out of date, try `force_reload=True` or see https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading for help.`
I tried using force_reload, the result is the same.

**Comment by glenn-jocher** [comment]
@JonathanSamelson regarding the loggers, not sure what the problem could be. The current logging code is here, with LOGGER imported in various places:
https://github.com/ultralytics/yolov5/blob/b94b59e199047aa8bf2cdd4401ae9f5f42b929e6/utils/general.py#L77-L88

Regarding the PyTorch v6.1 PyTorch Hub usage this works correctly for me, I'm not able to reproduce any errors:
<img width="1166" alt="Screenshot 2022-03-11 at 18 05 36" src="https://user-images.githubusercontent.com/26833433/157913899-10ec405f-210f-42bf-b6bf-4511ef995dd3.png">

**Comment by JonathanSamelson** [comment]
@glenn-jocher I think the difference in the format is caused by this line:

`logging.basicConfig(format="%(message)s", level=logging.INFO if (verbose and rank in (-1, 0)) else logging.WARNING)`

I think it causes the logger to change the format for all loggers instead of only yolov5 logger.

And actually I was wrong in my previous message, the logger does not stop working but the level of the logger changes to INFO so the DEBUG message are filtered.

**Comment by glenn-jocher** [comment]
@JonathanSamelson ok got it. If you have a fix in mind can you please submit a PR? Thanks!

**Comment by JonathanSamelson** [comment]
@glenn-jocher  At the moment, I don't. I'm using kind of the same lines in my project...
I understand it does not matter until it becomes an underlying project.

**Comment by glenn-jocher** [comment]
@arjitkatare @JonathanSamelson good news 😃! Your original issue may now be fixed ✅ in PR #7296 by @maxstrobel. 

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
